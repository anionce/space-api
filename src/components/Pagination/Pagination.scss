@import '../../assets/styles/variables';

.pagination {
	margin-top: 36px;
	display: flex;
	gap: 16px;

	.page:hover {
		cursor: pointer;
	}

	.active {
		font-weight: 800;
		text-decoration: underline $color-vidid;
		text-decoration-thickness: 5px;
	}
}
