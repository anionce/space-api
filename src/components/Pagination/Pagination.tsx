import React from 'react';
import './Pagination.scss';

type PaginationProps = {
	count: number | undefined;
	paginate: (arg: number) => void;
	pagesPerPage: number;
	currentPage: number;
};

export const Pagination = ({ count, paginate, pagesPerPage, currentPage }: PaginationProps) => {
	const hasItemsToShow: Boolean = Boolean(count);
	const pagesCount: number = Math.ceil((count || 25) / pagesPerPage);
	const onPageNumberClick = (i: number): void => {
		paginate(i);
	};

	return (
		<>
			{hasItemsToShow && (
				<div className='pagination'>
					{[...new Array(pagesCount)].map((_, i) => (
						<div
							title='pageNumber'
							key={i}
							className={i + 1 === currentPage ? 'active' : 'page'}
							onClick={() => onPageNumberClick(i + 1)}>
							{i + 1}
						</div>
					))}
				</div>
			)}
		</>
	);
};
