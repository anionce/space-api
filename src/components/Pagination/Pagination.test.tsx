import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { customRender } from '../../assets/tests/utils';
import { Pagination } from './Pagination';

const defaultProps = { count: 27, paginate: jest.fn(), pagesPerPage: 3, currentPage: 2 };

test('renders correcly', () => {
	const { container } = customRender(<Pagination {...defaultProps} />);

	expect(container).toMatchSnapshot();
});

test('active page has the correct class', () => {
	customRender(<Pagination {...defaultProps} />);

	expect(screen.getByText('2')).toHaveClass('active');
});

test('other pages have the correct class', () => {
	customRender(<Pagination {...defaultProps} />);

	expect(screen.getByText('3')).toHaveClass('page');
});

test('displays 9 pages', () => {
	customRender(<Pagination {...defaultProps} />);

	expect(screen.getAllByTitle(/pageNumber/i)).toHaveLength(9);
});
