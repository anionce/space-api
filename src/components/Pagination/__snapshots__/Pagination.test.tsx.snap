// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcly 1`] = `
<div>
  <div
    class="pagination"
  >
    <div
      class="page"
      title="pageNumber"
    >
      1
    </div>
    <div
      class="active"
      title="pageNumber"
    >
      2
    </div>
    <div
      class="page"
      title="pageNumber"
    >
      3
    </div>
    <div
      class="page"
      title="pageNumber"
    >
      4
    </div>
    <div
      class="page"
      title="pageNumber"
    >
      5
    </div>
    <div
      class="page"
      title="pageNumber"
    >
      6
    </div>
    <div
      class="page"
      title="pageNumber"
    >
      7
    </div>
    <div
      class="page"
      title="pageNumber"
    >
      8
    </div>
    <div
      class="page"
      title="pageNumber"
    >
      9
    </div>
  </div>
</div>
`;
